frontend/src/app/AppRouter.tsx

```diff
@@ -0,0 +1,28 @@
+import { lazy, Suspense } from 'react'
+import { Route, Switch } from 'react-router-dom'
+
+import { LOGIN_ROUTE, ROOT_ROUTE } from '~constants/routes'
+
+import { PrivateRoute } from './PrivateRoute'
+import { PublicRoute } from './PublicRoute'
+
+const WorkspacePage = lazy(() => import('~pages/workspace'))
+const LoginPage = lazy(() => import('~pages/login'))
+
+export const AppRouter = (): JSX.Element => {
+  return (
+    <Suspense fallback={<div>Loading...</div>}>
```

Comment:
is it wise to put a single suspense boundary so high up in the app? doesn't this mean that if we fail to wrap a lower-level lazy component in its own suspense boundary, the entire app will fallback to this div?

also don't we need a more deliberate design for the loading state?

also what suspense integration do we plan to use?

Reply:
I think it's fine, since this lazy-load distinguishes between the authenticated and non-authenticated parts of the application.
But yes, you are correct if we use another lazy loaded component it will fallback to this div. This div is also temporary, probably need @pearlyong to design a catch-all loading screen :P